pkg/pillar/cmd/volumemgr/handledownloader.go

```diff
@@ -110,6 +140,10 @@ func handleDownloaderStatusModify(ctxArg interface{}, key string,
 	if config == nil && status.RefCount == 0 {
 		log.Infof("handleDownloaderStatusModify adding RefCount=0 config %s\n",
 			key)
+
+		// where should the final downloaded file be?
+		locFilename := path.Join(types.DownloadDirname, status.ObjType, "pending", status.ImageID.String(), path.Base(status.Name))
```

Comment:
Different target than above.

Reply:
Is it? Here are the two:

```go
// above
locFilename := path.Join(types.DownloadDirname, status.ObjType, "pending", status.VolumeID.String(), path.Base(name))
// here
locFilename := path.Join(types.DownloadDirname, status.ObjType, "pending", status.ImageID.String(), path.Base(status.Name))
```

The only differences are in the last two parts. The second one (the one referenced here) is taking its arg as

```go
status := statusArg.(types.DownloaderStatus)
```

Whose `status.ImageID` was set in downloader from `config.ImageID`, which was set earlier in volumemgr from `volumeStatus.VolumeID`, so it all is the same. Confusing, but the same.

The last part is the name, which was determined above as:

```go
		name := status.DownloadOrigin.Name
```

and then set on the downloader config:

```go
		n := types.DownloaderConfig{
			ImageID:     status.VolumeID,
			DatastoreID: status.DownloadOrigin.DatastoreID,
			// XXX StorageConfig.Name is what?
			Name:        name, // XXX URL? DisplayName?
```

and then saved to `downloaderStatus.Name`, so it should be the same as well.

Reply:
Once the original bug in using VolumeID is fixed presumably these pathnames should both use imageID.

Reply:
Got it. So this is dependent on that. Once this passes review and tests (I will remove `WIP` to indicate both), then we can merge it in and fix it after the original bug is fixed. If that gets fixed first, we get it here too.

Reply:
Yep